src/foam/dao/MongoDAO.java

```diff
@@ -0,0 +1,176 @@
+    Subscription sub       = new Subscription();
+
+    String collectionName = (String) x.get("collectionName");
+    String collectionClass = (String) x.get("collectionClass");
+    List props = getFObjectProperties(collectionClass, x);
+
+    if ( collectionName == null || collectionName.isEmpty() ) {
+      throw new IllegalArgumentException("Invalid collection name in context. Please provide a string name.");
+    } else if ( collectionClass == null || collectionClass.isEmpty() ) {
+      throw new IllegalArgumentException("Invalid collection class name in context. Please provide a string class name.");
+    }
+
+    MongoCollection<Document> collection = database.getCollection(collectionName);
+    MongoCursor<Document> cursor = collection.find().iterator();
+
+    try {
+      while ( cursor.hasNext() ) {
+        if ( sub.getDetached() ) break;
+
+        FObject obj = createFObject(collectionClass, props, cursor.next());
+
+        if ( ( predicate == null ) || predicate.f(obj) ) {
+          decorated.put(obj, sub);
+        }
+      }
+    } finally {
+      cursor.close();
+    }
+
+    decorated.eof();
+
+    return sink;
+  }
+
+  private List getFObjectProperties(String clsName, X x) {    
+    try {
+      Class cls = Class.forName(clsName);
+      FObject clsInstance = (FObject) x.create(cls);
+      List props = clsInstance.getClassInfo().getAxiomsByClass(foam.core.AbstractFObjectPropertyInfo.class);
+      
+      // Recursively handle nested FObjectProperties
+      for ( int i = 0 ; i < props.size() ; i++ ) {
+        props.addAll(getFObjectProperties(((PropertyInfo) props.get(i)).getPropertyType(), x));
+      }
+
+      return props;
+    } catch (Exception ex) {
+      throw new RuntimeException(clsName + " was not found.");
+    }
+  }
+
+  private FObject createFObject(String clsName, List props, Document d) {
+    JsonWriterSettings writerSettings = new JsonWriterSettings(JsonMode.SHELL, true);
+    String jsonStr = d.toJson(writerSettings);
+
+    // Trims initial `"_id" : ObjectId("[24 HEX Chars]"),`
+    jsonStr = "{ class: \"" + clsName + "\", " + 
+                 jsonStr.substring(MONGO_OBJECT_PREFIX_LENGTH, jsonStr.length() - 1) + 
+              " }";
+
+    Iterator i = props.iterator();
+
+    while ( i.hasNext() ) {
+      PropertyInfo prop = (PropertyInfo) i.next();
+      jsonStr = jsonStr.replace("\"" + prop.getName() + "\" : {", "\"" + prop.getName() + "\" : { class: \"" + prop.getPropertyType() + "\",");
+    }
+
+    JSONParser parser = new JSONParser();
+    parser.setX(EmptyX.instance());
+    return parser.parseString(jsonStr);
+  }
+
+  @Override
+  public FObject put_(X x, FObject obj) {
+    throw new UnsupportedOperationException("put_ is not yet supported for MongoDAO");
+  }
+
+  @Override
+  public FObject find_(X x, Object id) {
+    throw new UnsupportedOperationException("find_ is not yet supported for MongoDAO");
```

Comment:
It might be easier to start with put_ and find_ together rather than select_(), that way you can iterate on proper serialization for a single object to/from MongoDB, after which you should be able to add enumeration without too much effort.